src/storage/v2/replication/replication_client.cpp

```diff
@@ -76,13 +76,37 @@ void ReplicationStorageClient::UpdateReplicaState(Storage *storage, DatabaseAcce
     }
   }
   if (branching_point) {
-    spdlog::error(
-        "You cannot register Replica {} to this Main because at one point "
-        "Replica {} acted as the Main instance. Both the Main and Replica {} "
-        "now hold unique data. Please resolve data conflicts and start the "
-        "replication on a clean instance.",
-        client_.name_, client_.name_, client_.name_);
-    replica_state_.WithLock([](auto &val) { val = replication::ReplicaState::DIVERGED_FROM_MAIN; });
+    replica_state_.WithLock([&](auto &state) { state = replication::ReplicaState::DIVERGED_FROM_MAIN; });
+    auto log_error = [client_name = client_.name_]() {
+      spdlog::error(
+          "You cannot register Replica {} to this Main because at one point "
+          "Replica {} acted as the Main instance. Both the Main and Replica {} "
+          "now hold unique data. Please resolve data conflicts and start the "
+          "replication on a clean instance.",
+          client_name, client_name, client_name);
+    };
+#ifdef MG_ENTERPRISE
+    if (!FLAGS_coordinator_server_port) {
+      log_error();
+      return;
+    }
+    replica_state_.WithLock([&](auto &state) {
+      state = replication::ReplicaState::DIVERGED_FROM_MAIN;
+      client_.thread_pool_.AddTask([&state, storage, gk = std::move(db_acc), this] {
+        if (!this->ForceResetStorage(storage)) {
+          spdlog::error(
+              "You cannot register REPLICA {} to this MAIN because MAIN couldn't reset REPLICA's storage."
+              "Please resolve data conflicts and start the "
+              "replication on a clean instance.",
+              client_.name_);
+          return;
+        }
+        state = replication::ReplicaState::RECOVERY;
+        this->RecoverReplica(0, storage);
```

Comment:
This should be executed in via the client's thread pool
```
client_.thread_pool_.AddTask([storage, current_commit_timestamp, gk = std::move(db_acc), this] {
    this->RecoverReplica(current_commit_timestamp, storage);
});
```

Reply:
It is already

Reply:
this is already in a task